### PR TITLE
OSPP 2021: Wechaty Lark Puppet

### PR DESCRIPTION
**Pull request by remember00000**
Please make sure to check the following boxes creating your pull request, thank you very much!

 Add Blog Header, including title, author, image, categories, tags, etc.

For example:

---
title: Your Great Blog Title
author: your_github_id
categories: project
tags:
  - python
  - padlocal
image: /assets/2021/02-your-blog-slug/teaser.jpg
---

One line bio

Your blog post content Your blog post content
Your blog post content Your blog post content
Your blog post content Your blog post content

It will be easier to start drafting your post by using a existing post file as your template, for example, this one

 Keep all filenames lowercase, and use - to connect words instead of space. e.g. 2017-10-06-wechat-pc-impactor.md instead of 2017-10-06-WeChat PC Impactor.md

 Embed video/pdf from YouTube/Bilibili/local (if there's any) by using {% include iframe.html src="URL_or_PATH" %}. (Learn more about how to embed file/url in the post by reading this blog post

 Save photo to local before publishing, to a folder named by ${MONTH}-${YOUR_BLOG_SLUG}/ under /assets/2021/ directory.

 Select a beautiful and meaningful teaser image for your blog

 Create your contributor profile (if you are a first time contributor)

For excample:

jekyll/_contributors/lijiarui.md

---
name: Jiarui LI (李佳芮)
site: https://lijiarui.github.io
bio: 句子互动创始人 & CEO，微软人工智能最具价值专家 (AI MVP)
avatar: /jekyll/assets/contributors/lijiarui/avatar.png
email: rui@juzi.bot
twitter: https://weibo.com/u/2175505900
---

Here is the example file:

lijiarui.md: https://github.com/wechaty/wechaty.js.org/blob/master/jekyll/_contributors/lijiarui.md
It will shown like this on wechaty website: https://wechaty.js.org/contributors/lijiarui/
To learn more about the contribute guideline, visit: https://wechaty.js.org/docs/contributor-program/

### CODE REVIEW

**Review by lijiarui — APPROVED** [review]
Thanks for the post!

Please:
1. Sign CLA first
2. Make GitHub action green. You can run `npm run test` locally to make it pass the test.
3. You can ask your mentor help you to review the blog first. When your mentor approve the blog, I will review your blog.

File: jekyll/_posts/2021-07-14-wechaty-puppet-lark-plan-blog.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  - [ ] 实现之前未实现的函数
+  - [ ] 连接Contact、Message、Room等类，尝试实现复用
+  - [ ] 撰写文档、example
+  - [ ] 发布npm包
+
+- 计划安排
+
+  - 熟悉wechaty
+    - 7.14 - 7.17
+    - 使用wecharty
```
Comment (Roxanne718) [inline jekyll/_posts/2021-07-14-wechaty-puppet-lark-plan-blog.md]: 'Wechaty' here, not 'wecharty'

Comment (remember00000) [reply jekyll/_posts/2021-07-14-wechaty-puppet-lark-plan-blog.md]: done~

**Comment by Roxanne718** [comment]
LGTM

**Comment by lijiarui** [comment]
Hi @remember00000 Please make Github action green, then I will review this PR.

You can run `npm run test` locally to make it pass the test. 

@Roxanne718 Could you help @remember00000 to make GitHub action green? thanks!

**Comment by remember00000** [comment]
> Hi @remember00000 Please make Github action green, then I will review this PR.
> 
> You can run `npm run test` locally to make it pass the test.
> 
> @Roxanne718 Could you help @remember00000 to make GitHub action green? t

**Comment by remember00000** [comment]
> Hi @remember00000 Please make Github action green, then I will review this PR.
> 
> You can run `npm run test` locally to make it pass the test.
> 
> @Roxanne718 Could you help @remember00000 to make GitHub action green? thanks!

I try this,but I met something and I can solve that by google;So I ask in Issues,and I hope to run npm run test locally successfully.thanks

**Comment by lijiarui** [comment]
Hi @remember00000 As [Github Action shown](https://github.com/wechaty/wechaty.js.org/pull/1085/checks?check_run_id=3089103272) Your error is related to the image issues, you can read the error info and change as the error required.

```
ok 18 total 197 files checked.
# developer project avatar should be put under assets/contributors/ folder
not ok 19 jekyll/assets/contributors/remember00000/avatar.jpg should exist
  ---
    operator: fail
    at: Test.<anonymous> (/home/runner/work/wechaty.js.org/wechaty.js.org/tests/jekyll/contributor-profile.spec.ts:91:11)
    stack: |-
      Error: jekyll/assets/contributors/remember00000/avatar.jpg should exist
          at Test.assert [as _assert] (/home/runner/work/wechaty.js.org/wechaty.js.org/node_modules/tape/lib/test.js:228:54)
          at Test.bound [as _assert] (/home/runner/work/wechaty.js.org/wechaty.js.org/node_modules/tape/lib/test.js:80:32)
          at Test.fail (/home/runner/work/wechaty.js.org/wechaty.js.org/node_modules/tape/lib/test.js:322:10)
          at Test.bound [as fail] (/home/runner/work/wechaty.js.org/wechaty.js.org/node_modules/tape/lib/test.js:80:32)
          at Test.<anonymous> (/home/runner/work/wechaty.js.org/wechaty.js.org/tests/jekyll/contributor-profile.spec.ts:91:11)
  ...
ok 20 total 197 files checked
# developer profile name must be GitHub username
.......
ok 21 7 contributors profile names checked
# misplaced files
ok 22 should no miss placed files. 
# folder _contributors/, _posts/, and assets/ has been moved to `jekyll/` (e.g. _posts/ => jekyll/_posts/)
ok 23 7 deprecated folders checked.
# filename only allow [a-z0-9-_.]
ok 24 total 2252 files checked.
# front matter key `image` must has a value to define the teaser image
ok 25 total 328 files checked
# {% include iframe.html src=... %} should exist in assets/ folder
ok 26 total 328 files checked.
# all remote images linked from the post should be exist.
.
ok 27 total 1 remote image urls checked
# all local images linked from the post should be exist.
not ok 28 "/assets/2021/07-wechaty-puppet-lark-plan-blog/wechaty-lark.png" should exist
  ---
    operator: fail
    at: Test.<anonymous> (/home/runner/work/wechaty.js.org/wechaty.js.org/tests/jekyll/image-existance.spec.ts:104:11)
    stack: |-
      Error: "/assets/2021/07-wechaty-puppet-lark-plan-blog/wechaty-lark.png" should exist
          at Test.assert [as _assert] (/home/runner/work/wechaty.js.org/wechaty.js.org/node_modules/tape/lib/test.js:228:54)
          at Test.bound [as _assert] (/home/runner/work/wechaty.js.org/wechaty.js.org/node_modules/tape/lib/test.js:80:32)
          at Test.fail (/home/runner/work/wechaty.js.org/wechaty.js.org/node_modules/tape/lib/test.js:322:10)
          at Test.bound [as fail] (/home/runner/work/wechaty.js.org/wechaty.js.org/node_modules/tape/lib/test.js:80:32)
          at Test.<anonymous> (/home/runner/work/wechaty.js.org/wechaty.js.org/tests/jekyll/image-existance.spec.ts:104:11)
  ...
not ok 29 "/assets/contributors/remember00000/avatar.jpg" should exist
  ---
    operator: fail
    at: Test.<anonymous> (/home/runner/work/wechaty.js.org/wechaty.js.org/tests/jekyll/image-existance.spec.ts:104:11)
    stack: |-
      Error: "/assets/contributors/remember00000/avatar.jpg" should exist
          at Test.assert [as _assert] (/home/runner/work/wechaty.js.org/wechaty.js.org/node_modules/tape/lib/test.js:228:54)
          at Test.bound [as _assert] (/home/runner/work/wechaty.js.org/wechaty.js.org/node_modules/tape/lib/test.js:80:32)
          at Test.fail (/home/runner/work/wechaty.js.org/wechaty.js.org/node_modules/tape/lib/test.js:322:10)
          at Test.bound [as fail] (/home/runner/work/wechaty.js.org/wechaty.js.org/node_modules/tape/lib/test.js:80:32)
          at Test.<anonymous> (/home/runner/work/wechaty.js.org/wechaty.js.org/tests/jekyll/image-existance.spec.ts:104:11)
  ...
```

As you mention in #1093 It seems this error is totally different from the issue you report, if you can give reproduce code, I would like to help.

File: jekyll/_posts/2021-07-14-wechaty-puppet-lark-plan-blog.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
+---
+ title: "基于开放 API 封装 Wechaty 接口下的飞书聊天机器人:期初具体计划"
+ author: remember00000
+ categories: project
```
Comment (lijiarui) [inline jekyll/_posts/2021-07-14-wechaty-puppet-lark-plan-blog.md]: Please change to the following as [OSPP Guide](https://wechaty.js.org/2021/06/24/summer-wechaty-guide/) required.
```
categories:
  - project
  - ospp
```

File: jekyll/_posts/2021-07-14-wechaty-puppet-lark-plan-blog.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ tags:
+  - summer-2021
+  - puppet-provider
+  - lark
+  - summer-of-wechaty
+  - ecosystem
```
Comment (lijiarui) [inline jekyll/_posts/2021-07-14-wechaty-puppet-lark-plan-blog.md]: Please add the following tag as [OSPP Guide](https://wechaty.js.org/2021/06/24/summer-wechaty-guide/) required:

`ospp`,`ospp-2021`,`plan`

File: jekyll/_posts/2021-07-14-wechaty-puppet-lark-plan-blog.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+本项目 [基于开放 API 封装 Wechaty 接口下的飞书聊天机器人] 系 暑期2021 支持的开源项目。
+
+<!--more-->
+
+## [基于开放 API 封装 Wechaty 接口下的飞书聊天机器人]具体计划
+
+- 导师：范蕊
+
+- 学生：马田慧
+
```
Comment (lijiarui) [inline jekyll/_posts/2021-07-14-wechaty-puppet-lark-plan-blog.md]: Please add the project link here: https://github.com/wechaty/summer-of-wechaty/issues/38

**Review by lijiarui — CHANGES_REQUESTED** [review]
Thanks for making Github Action green!

Please change as I request, and it would be better if you can refer   https://github.com/wechaty/wechaty.js.org/pull/1078/files to make the blog more beautiful/

File: jekyll/_posts/2021-07-14-wechaty-puppet-lark-plan-blog.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
+---
+ title: "基于开放 API 封装 Wechaty 接口下的飞书聊天机器人:期初具体计划"
```
Comment (lijiarui) [inline jekyll/_posts/2021-07-14-wechaty-puppet-lark-plan-blog.md]: Please change title to `OSPP 2021-期初报告-你的标题` as https://wechaty.js.org/2021/06/24/summer-wechaty-guide/ required.

**Review by lijiarui — CHANGES_REQUESTED** [review]
Please change as I request. 

Also, please change the file name from `jekyll/_posts/2021-07-14-wechaty-puppet-lark-plan-blog.md` to `2021-07-14-ospp-plan-XX`as https://wechaty.js.org/2021/06/24/summer-wechaty-guide/ required.

**Comment by lijiarui** [comment]
@remember00000 Please make Github action green.

**Comment by lijiarui** [comment]
Thank you very much for your contribution!

## You are welcome to join [Wechaty Contributor Program](https://wechaty.js.org/docs/contributor-program/)

### 1. Join Wechaty Organization

> You've invited  remember00000 to Wechaty! They'll be receiving an email shortly. They can also visit https://github.com/wechaty to accept the invitation.

I have invited you to join our Wechaty GitHub Organization, please accept it by following the above message.

### 2. Update Your Wechaty Contributor Profile

1. Please open [Contributor Hall of Fame](https://docs.google.com/spreadsheets/d/1XcDoIczyIclqXP1p90Sz7S0n4Q22xjFzJmjPFlU2g1E/edit#gid=0) and add yourself to the end of the list, so that other contributors will know you better!
1. Please add yourself to our Website [Contributors](https://wechaty.js.org/contributors/) by [creating a PR](https://github.com/wechaty/wechaty.js.org/tree/master/jekyll/_contributors) and refer to this PR link as well.

### 3. Join The Contributor Only WeChat Room

We also have a WeChat room for contributors only which can discuss Wechaty at a deeper level, you are welcome to join and if you are interested.

Please add my: wechat: _ruirui_0914_ and send this pr link to me. I will invite you into Wechaty Contributor Room

Link to https://github.com/wechaty/PMC/issues/16

Cheers!

**Pull request merged**